src/controller/AutoCommissioner.cpp

```diff
@@ -504,12 +533,33 @@ CHIP_ERROR AutoCommissioner::CommissioningStepFinished(CHIP_ERROR err, Commissio
         return CHIP_ERROR_INCORRECT_STATE;
     }
 
-    mParams.SetCompletionStatus(completionStatus);
     mCommissioner->PerformCommissioningStep(proxy, nextStage, mParams, this, GetEndpoint(nextStage),
                                             GetCommandTimeout(proxy, nextStage));
     return CHIP_NO_ERROR;
 }
 
+void AutoCommissioner::PauseCommissioning()
+{
+    mCommissioningPaused = true;
+}
+
+CHIP_ERROR AutoCommissioner::ResumeCommissioning()
+{
+    VerifyOrReturnError(mCommissioningPaused, CHIP_ERROR_INCORRECT_STATE);
+    mCommissioningPaused = false;
+
+    // if no new step was attempted
+    if (mPausedStage == CommissioningStage::kError)
+    {
+        return CHIP_NO_ERROR;
+    }
+
+    CommissioningStage nextStage = mPausedStage;
+    mPausedStage                 = CommissioningStage::kError;
+
+    return PerformStep(nextStage);
```

Comment:
If at this point we don't have the right proxy, this will just return error instead of doing the cleanup the caller of `CommissioningStepFinished` would do on error.  Is that OK because the only way we could be in that state is if the cleanup has already happened?  Might be worth documenting that here.

Reply:
I added GetDeviceProxyForStep private method and now call it before exiting due to a paused state so that the caller can cleanup in this error case.